### PR TITLE
DOC Enable `.to(device)` semantics for pytorch examples

### PR DESCRIPTION
**Pull request by MuawizChaudhary**
This PR enables `to(device)` semantics for the pytorch examples. 

This resolves issue #824 .

### CODE REVIEW

**Comment by MuawizChaudhary** [comment]
We should really merge #829 , #826  before this.

File: examples/1d/reconstruct_torch.py [hunk 1]
```diff
@@ -58,16 +59,16 @@ def generate_harmonic_signal(T, num_intervals=4, gamma=0.9, random_state=42):
 # Let’s take a look at what such a signal could look like.
 
 T = 2 ** 13
-x = torch.from_numpy(generate_harmonic_signal(T))
+x = torch.from_numpy(generate_harmonic_signal(T)).to(device)
```
Comment (janden) [inline examples/1d/reconstruct_torch.py:62]: Maybe just move this call to just before the `scattering` call (so have `x = x.to(device)` on the line before)?

File: examples/2d/cifar_resnet_torch.py [hunk 2]
```diff
@@ -158,20 +158,18 @@ def test(model, device, test_loader, scattering):
...
 
     model = Scattering2dResNet(K, args.width).to(device)
 
     # DataLoaders
+    num_workers = 4
     if use_cuda:
-        num_workers = 4
         pin_memory = True
     else:
-        num_workers = None
```
Comment (janden) [inline examples/2d/cifar_resnet_torch.py:174]: ?

**Review by janden — CHANGES_REQUESTED** [review]
Wonderful!

File: examples/2d/cifar_resnet_torch.py (hunk 2 above)
Comment (MuawizChaudhary) [reply examples/2d/cifar_resnet_torch.py:174]: Why should number of cpu workers change depending on if we're on cpu or not?

**Comment by MuawizChaudhary** [comment]
looking at the commit history... did i rebase all wrong Q_Q

File: examples/2d/cifar_resnet_torch.py (hunk 2 above)
Comment (janden) [reply examples/2d/cifar_resnet_torch.py:174]: Agreed, but this was maybe there for a reason. Does it affect performance?

Comment (MuawizChaudhary) [reply examples/2d/cifar_resnet_torch.py:174]: `num_workers = None` appears to give me an [error](https://colab.research.google.com/drive/10_bgL7GRPvrpLiMBxd__aK1mtdj3U4wr#scrollTo=0Pp33Tr7qZIM). 

I don't think there are many reasons to set `num_works = 0 ` when most modernday cpus are multicore. @eugenium care to chime in?

Comment (janden) [reply examples/2d/cifar_resnet_torch.py:174]: Ok.

**Review by janden — APPROVED** [review]

**Comment by janden** [comment]
Nm looks like it's the TF FFT implementation which is still flaky.

**Pull request merged**